lib/block-supports/elements.php

```diff
@@ -87,35 +87,26 @@ function gutenberg_render_elements_support( $block_content, $block ) {
  * @return null
  */
 function gutenberg_render_elements_support_styles( $pre_render, $block ) {
-	$block_type           = WP_Block_Type_Registry::get_instance()->get_registered( $block['blockName'] );
-	$element_block_styles = isset( $block['attrs']['style']['elements'] ) ? $block['attrs']['style']['elements'] : null;
-
-	/*
-	* For now we only care about link color.
-	* This code in the future when we have a public API
-	* should take advantage of WP_Theme_JSON_Gutenberg::compute_style_properties
-	* and work for any element and style.
-	*/
+	$block_type                    = WP_Block_Type_Registry::get_instance()->get_registered( $block['blockName'] );
+	$element_block_styles          = isset( $block['attrs']['style']['elements'] ) ? $block['attrs']['style']['elements'] : null;
 	$skip_link_color_serialization = gutenberg_should_skip_block_supports_serialization( $block_type, 'color', 'link' );
 
-	if ( $skip_link_color_serialization ) {
+	if ( empty( $element_block_styles ) || $skip_link_color_serialization ) {
```

Comment:
When we start introducing more elements and element styles, we'll have to rebuild each element with "skip serialization" checks for all their style properties.

Reply:
Yeh I noticed. I'd like to understand more about why link color serlization is potentially skipped. What's the deal there?

Reply:
It's a block supports feature introduced a while back. With it, you can specify at the block.json level whether a block receives a particular style. 

See:

- https://github.com/WordPress/gutenberg/pull/36293